src/Microsoft.ML.LightGBM/LightGbmArguments.cs

```diff
@@ -314,98 +431,184 @@ public enum EvalMetricType
         public int MaximumBinCountPerFeature = 255;
 
+        /// <summary>
+        /// Determines which booster to use.
+        /// </summary>
+        /// <value>
+        /// Available boosters are <see cref="DartBooster"/>, <see cref="GossBooster"/>, and <see cref="TreeBooster"/>.
+        /// </value>
         [Argument(ArgumentType.Multiple, HelpText = "Which booster to use, can be gbtree, gblinear or dart. gbtree and dart use tree based model while gblinear uses linear function.", SortOrder = 3)]
         public ISupportBoosterParameterFactory Booster = new TreeBooster.Options();
 
+        /// <summary>
+        /// Determines whether to output progress status during training and evaluation.
+        /// </summary>
         [Argument(ArgumentType.AtMostOnce, HelpText = "Verbose", ShortName = "v")]
         public bool Verbose = false;
 
+        /// <summary>
+        /// Controls the logging level in LighGBM.
+        /// </summary>
+        /// <value>
+        /// <see langword="true"/> means only output Fatal errors. <see langword="false"/> means output Fatal, Warning, and Info level messages.
+        /// </value>
         [Argument(ArgumentType.AtMostOnce, HelpText = "Printing running messages.")]
         public bool Silent = true;
 
+        /// <summary>
+        /// Determines the number of threads used to run LightGBM.
+        /// </summary>
         [Argument(ArgumentType.AtMostOnce, HelpText = "Number of parallel threads used to run LightGBM.", ShortName = "nt")]
         public int? NumberOfThreads;
 
+        /// <summary>
+        /// Determines what evaluation metric to use.
+        /// </summary>
         [Argument(ArgumentType.AtMostOnce,
             HelpText = "Evaluation metrics.",
             ShortName = "em")]
         public EvalMetricType EvaluationMetric = EvalMetricType.DefaultMetric;
 
+        /// <summary>
+        /// Whether to use softmax loss. Used only by <see cref="LightGbmMulticlassTrainer"/>.
+        /// </summary>
         [Argument(ArgumentType.AtMostOnce, HelpText = "Use softmax loss for the multi classification.")]
         [TlcModule.SweepableDiscreteParam("UseSoftmax", new object[] { true, false })]
         public bool? UseSoftmax;
 
+        /// <summary>
+        /// Determines the number of rounds, after which training will stop if validation metric doesn't improve.
+        /// </summary>
+        /// <value>
+        /// 0 means disable early stopping.
+        /// </value>
         [Argument(ArgumentType.AtMostOnce, HelpText = "Rounds of early stopping, 0 will disable it.",
             ShortName = "es")]
         public int EarlyStoppingRound = 0;
 
-        [Argument(ArgumentType.AtMostOnce, HelpText = "Comma seperated list of gains associated to each relevance label.", ShortName = "gains")]
+        /// <summary>
+        /// Comma separated list of gains associated with each relevance label. Used only by <see cref="LightGbmRankingTrainer"/>.
```

Comment:
>a sep [](start = 16, length = 5)

hyphen

Reply:
done

---
In reply to: [263946259](https://github.com/dotnet/machinelearning/pull/2886#discussion_r263946259) [](ancestors = 263946259)